src/Streamly/Internal/LZ4.hs

```diff
@@ -398,7 +420,20 @@ resizeD Config{..} (Stream.Stream step0 state0) =
                 arr1 <- Array.spliceTwo buf arr
                 liftIO $ process st1 arr1
             Stream.Skip st1 -> return $ Stream.Skip $ RAccumlate st1 buf
-            Stream.Stop -> return $ Stream.Skip $ RYield buf RDone
+            Stream.Stop -> error "resizeD: Incomplete block"
+    step gst (RFooter st buf) = do
+        r <- step0 gst st
+        case r of
+            Stream.Yield arr st1 -> do
+                arr1 <- Array.spliceTwo buf arr
+                return $ Stream.Skip $ RFooter st1 arr1
```

Comment:
We should not continue indefinitely and wait for the stream to stop before we validate the footer. As soon as we have got enough data we should validate and stop. In case we detect that there is more data after it then we can emit a warning.

Reply:
Right now I don't emit any warning. We don't endefinately consume though.

Reply:
`footerSize` in the comfig is a general and may not be required but we can keep it for now.